cmd/argocd-applicationset-controller/commands/applicationset_controller.go

```diff
@@ -131,6 +132,7 @@ func NewCommand() *cobra.Command {
 				startWebhookServer(webhookHandler, webhookAddr)
 			}
 			askPassServer := askpass.NewServer()
+			go func() { errors.CheckError(askPassServer.Run(askpass.SocketPath)) }()
```

Comment:
The PR looks good to me. Since `askPassServer.Run(path string)` accepts path as an argument. Wouldn't it be easy to pass different paths for both repo server and applicationset? Currently, the path would be `/tmp/reposerver-ask-pass.sock` for the applicationset unless explicitly changed by an env variable. Just thinking out loud...

Reply:
Thanks for the explanation @rishabh625. It makes sense now. Without an env/flag, the path will be fixed when the `argocd-git-ask-pass` command will be invoked.